databricks/koalas/series.py

```diff
@@ -4335,6 +4335,80 @@ def pct_change(self, periods=1):
 
         return self._with_new_scol((scol - prev_row) / prev_row)
 
+    def dot(self, other):
+        """
+        Compute the dot product between the Series and the columns of other.
+
+        This method computes the dot product between the Series and another
+        one, or the Series and each columns of a DataFrame.
+
+        It can also be called using `self @ other` in Python >= 3.5.
+
+        .. note:: The behaviour of this API is slightly different from pandas
+            since its original implementation requires quite expensive operation
+            (It requires sorting & comparing for all index values which is quite
+            expensive and may cause out-of-memory with large dataset).
+            pandas raises ValueError if all values of index are not same, while
+            Koalas just ignoring them as NaN.
```

Comment:
This API is slightly different from pandas when indexes from both are not aligned. To match with pandas', it requires to read the whole data for, for example, counting. pandas raises an exception; however, Koalas just proceeds and performs by ignoring mismatches with NaN permissively.